src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -80,11 +85,221 @@ static ShardPlacement * ShardPlacementOnGroup(uint64 shardId, int groupId);
 
 
 /* exports for SQL callable functions */
+PG_FUNCTION_INFO_V1(citus_node_disk_space_stats);
+PG_FUNCTION_INFO_V1(citus_local_disk_space_stats);
+PG_FUNCTION_INFO_V1(citus_database_size);
+PG_FUNCTION_INFO_V1(citus_node_database_size);
 PG_FUNCTION_INFO_V1(citus_table_size);
 PG_FUNCTION_INFO_V1(citus_total_relation_size);
 PG_FUNCTION_INFO_V1(citus_relation_size);
 
 
+/*
+ * citus_node_database_size is a UDF that returns the size of the database
+ * on a given node.
+ */
+Datum
+citus_node_disk_space_stats(PG_FUNCTION_ARGS)
+{
+	text *nodeNameText = PG_GETARG_TEXT_P(0);
+	char *nodeNameString = text_to_cstring(nodeNameText);
+	int32 nodePort = PG_GETARG_INT32(1);
+
+	CheckCitusVersion(ERROR);
+
+	uint64 availableBytes = 0;
+	uint64 totalBytes = 0;
+
+	GetNodeDiskSpaceStats(nodeNameString, nodePort, &availableBytes, &totalBytes);
+
+	TupleDesc tupleDescriptor = NULL;
+	Datum values[TABLE_METADATA_FIELDS];
+	bool isNulls[TABLE_METADATA_FIELDS];
+
+	TypeFuncClass resultTypeClass = get_call_result_type(fcinfo, NULL,
+														 &tupleDescriptor);
+	if (resultTypeClass != TYPEFUNC_COMPOSITE)
+	{
+		ereport(ERROR, (errmsg("return type must be a row type")));
+	}
+
+	/* form heap tuple for remote disk space statistics */
+	memset(values, 0, sizeof(values));
+	memset(isNulls, false, sizeof(isNulls));
+
+	values[0] = UInt64GetDatum(availableBytes);
+	values[1] = UInt64GetDatum(totalBytes);
+
+	HeapTuple diskSpaceTuple = heap_form_tuple(tupleDescriptor, values, isNulls);
+
+	PG_RETURN_UINT64(HeapTupleGetDatum(diskSpaceTuple));
+}
+
+
+/*
+ * GetNodeDiskSpaceStats fetches the disk space statistics for a given node,
+ * or returns false if unsuccessful.
+ */
+static bool
+GetNodeDiskSpaceStats(char *nodeName, int nodePort, uint64 *availableBytes,
+					  uint64 *totalBytes)
+{
+	uint32 connectionFlags = 0;
+	PGresult *result = NULL;
+	bool raiseErrors = true;
+
+	char *sizeQuery = "SELECT available_disk_size, total_disk_size "
+					  "FROM pg_catalog.citus_local_disk_space_stats()";
+
+	MultiConnection *connection = GetNodeConnection(connectionFlags, nodeName, nodePort);
+
+	int queryResult = ExecuteOptionalRemoteCommand(connection, sizeQuery, &result);
+	if (queryResult != 0)
```

Comment:
Minor: we can possibly combine this with the below check `queryResult != 0 || !IsReponseOk(result ....`